packages/optimizer/lib/transformers/PreloadHeroImage.js

```diff
@@ -78,6 +83,28 @@ class PreloadHeroImage {
     insertAfter(head, preload, referenceNode);
   }
 
+  hasExistingImagePreload(head) {
+    for (const node of head.children) {
+      if (this.isImagePreload(node)) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
continuing my series of "unhelpful comments that make the code less readable"...

```js
return [...head.children].any(this.isImagePreload)
```

Reply:
Why the destructuring? I think `head.children.some(this.isImagePreload);` does the trick as well (and is easy to read).